components/workflow.rst

```diff
@@ -19,6 +19,117 @@ You can install the component in 2 different ways:
+
+    $states = ['draft', 'review', 'rejected', 'published'];
+    $transactions[] = new Transition('to_review', ['draft', 'rejected'], 'review');
+    $transactions[] = new Transition('publish', 'review', 'published');
+    $transactions[] = new Transition('reject', 'review', 'rejected');
+
+    $definition = new Definition($states, $transactions);
+    $definition->setInitialPlace('draft');
+
+    $marking = new ScalarMarkingStore('currentState');
+    $workflow = new Workflow($definition, $marking);
+
+The ``Workflow`` can now help you to decide what actions that are allowed on a blog post.
+
+.. code-block:: php
+
+    $post = new \stdClass();
+    $post->currentState = null;
+    $workflow->can($post, 'publish'); // False
+    $workflow->can($post, 'to_draft'); // True
+
+    // Update the currentState on the post
+    try {
+        $workflow->apply($post, 'to_review');
+    } catch (LogicException $e) {
+        // ...
+    }
+
+    // See all the available transaction for the post in the current state
+    $transactions = $workflow->getEnabledTransitions($post);
+
+
+
+Using events
+------------
+
+To make your workflows even more powerful you could construct the ``Workflow`` object with an ``EventDispatcher``. You
+can now create event listeners to block transactions ie depending on the data in the blog post. The following events
+are dispatched:
+
+* workflow.guard
+* workflow.[workflow name].guard
+* workflow.[workflow name].guard.[transaction name]
+
+See example to make sure no blog post without title is moved to "review"::
+
+    $marking = new ScalarMarkingStore('currentState');
+    $workflow = new Workflow($definition, $marking, $dispatcher, 'blogpost');
+
+.. code-block:: php
+
+    class BlogPostReviewListener implements EventSubscriberInterface
+    {
+        public function guardReview(GuardEvent $event)
+        {
+            $post = $event->getSubject();
+            $title = $post->getTitle();
+
+            if (empty($title)) {
+                // Posts with no title should not be allowed
+                $event->setBlocked(true);
+            }
+        }
+
+        public static function getSubscribedEvents()
+        {
+            return array(
+                'workflow.blogpost.guad.to_review' => array('guardReview'),
+            );
+        }
+    }
+
+With help from the ``EventDispatcher`` and the ``AuditTrailListener`` you could easily enable logging::
+
+    $logger = new PSR3Logger();
+    $subscriber = new AuditTrailListener($logger);
+    $dispatcher->addSubscriber($subscriber);
+
+Dumper
+------
+
+To help you debug you could dump a representation of your workflow with the use of a ``DumperInterface``. Use the
+``GraphvizDumper`` to create a PNG image of the workflow defined above::
+
+    // dump-graph.php
+    $dumper = new GraphvizDumper();
+    echo $dumper->dump($definition);
+
+.. code-block:: bash
+
+    $ php dump-graph-php > out.dot
+    $ dot -Tpng out.dot -o graph.png
```

Comment:
You are very correct. I added a note about graphviz. Yes, is available for Linux, Solaris, Windows and Mac.